chia/wallet/puzzles/clawback/metadata.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import IntEnum
+from typing import List, Optional
+
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint16, uint64
+from chia.util.streamable import Streamable, streamable
+from chia.wallet.wallet_puzzle_store import WalletPuzzleStore
+
+
+@streamable
+@dataclass(frozen=True)
+class ClawbackMetadata(Streamable):
+    time_lock: uint64
+    sender_puzzle_hash: bytes32
+    recipient_puzzle_hash: bytes32
+    memos: Optional[List[bytes]] = None
+
+    async def is_recipient(self, puzzle_store: WalletPuzzleStore) -> bool:
+        if await puzzle_store.puzzle_hash_exists(self.sender_puzzle_hash):
```

Comment:
Yes, since the sender can clawback at any time. It doesn't make sense to ask the sender to wait a long time to retrieve their coins, if they send to themselves.